### PR TITLE
SAK-30464 Added unreadMessagesCount endpoint

### PR DESCRIPTION
**Pull request by adrianfish**

### CODE REVIEW

File: profile2/tool/src/java/org/sakaiproject/profile2/tool/entityprovider/ProfileEntityProvider.java [hunk 1]
```diff
@@ -251,7 +252,26 @@ public Object getConnectionStatus(EntityReference ref, Map<String, Object> param
...
+		if(StringUtils.isBlank(uuid)) {
+			throw new EntityNotFoundException("Invalid user.", ref.getId());
+		}
+        
+        if (!sakaiProxy.getCurrentUserId().equals(uuid)) {
+			throw new SecurityException("You can only view your own message count.");
+        }
+
+		return new ActionReturn(messagingLogic.getAllUnreadMessagesCount(uuid));
+	}
```
Comment (steveswinsburg) [inline profile2/tool/src/java/org/sakaiproject/profile2/tool/entityprovider/ProfileEntityProvider.java]: Looks good. Weird spacing though bro.

Comment (adrianfish) [inline profile2/tool/src/java/org/sakaiproject/profile2/tool/entityprovider/ProfileEntityProvider.java]: Bloody hard tabs. I'll rejig it.

Comment (adrianfish) [inline profile2/tool/src/java/org/sakaiproject/profile2/tool/entityprovider/ProfileEntityProvider.java]: I've just pushed an update, ta for looking at it.

File: profile2/tool/src/java/org/sakaiproject/profile2/tool/entityprovider/ProfileEntityProvider.java [hunk 2]
```diff
@@ -251,7 +252,26 @@ public Object getConnectionStatus(EntityReference ref, Map<String, Object> param
...
+			throw new SecurityException("You must be logged in to get the unread messages count.");
+		}
+
+		//convert input to uuid
+		String uuid = sakaiProxy.ensureUuid(ref.getId());
+		if(StringUtils.isBlank(uuid)) {
+			throw new EntityNotFoundException("Invalid user.", ref.getId());
+		}
+        
+		if (!sakaiProxy.getCurrentUserId().equals(uuid)) {
```
Comment (jonespm) [inline profile2/tool/src/java/org/sakaiproject/profile2/tool/entityprovider/ProfileEntityProvider.java]: I'd think if you're an admin user calling this, you could check any message count

Comment (adrianfish) [inline profile2/tool/src/java/org/sakaiproject/profile2/tool/entityprovider/ProfileEntityProvider.java]: I didn't think admins would be that interested, but, it's easy so ...

I'll rejig it :)

**Comment by adrianfish** [comment]
I've added the admin user check and tidied up some more tab inconsistencies.

**Pull request merged**